### PR TITLE
Fix test bug with extra small complex geometries

### LINKED ISSUE #93889: [CI] LatLonGeometryRelationVisitorTests testPoint failing
**Build scan:**
https://gradle-enterprise.elastic.co/s/5uv2agkmc7czs/tests/:x-pack:plugin:spatial:test/org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests/testPoint

**Reproduction line:**
```
./gradlew ':x-pack:plugin:spatial:test' --tests "org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.testPoint" -Dtests.seed=9459A78F7AA23686 -Dtests.locale=ga -Dtests.timezone=Pacific/Chatham -Druntime.java=17
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests&tests.test=testPoint

**Failure excerpt:**
```
java.lang.AssertionError: WITHIN/QUERY_INSIDE: Geometry Point(0.0,0.0) relates to shape Line: QUERY_INSIDE
Expected: <false>
     but: was <true>

  at __randomizedtesting.SeedInfo.seed([9459A78F7AA23686:A9068DC41E572BD5]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.doTestShape(LatLonGeometryRelationVisitorTests.java:202)
  at org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.doTestShape(LatLonGeometryRelationVisitorTests.java:163)
  at org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.doTestShape(LatLonGeometryRelationVisitorTests.java:158)
  at org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.doTestShapes(LatLonGeometryRelationVisitorTests.java:148)
  at org.elasticsearch.xpack.spatial.index.fielddata.LatLonGeometryRelationVisitorTests.testPoint(LatLonGeometryRelationVisitorTests.java:41)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

### PR DESCRIPTION
**Pull request by craigtaverner**
If the complex geometry contains only points that all resolve (quantize) to the same single point, then the entire geometry is identical to a single point, and the point and the geometry are both inside and contain each other.

Fixes #93889

### CODE REVIEW

File: x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/LatLonGeometryRelationVisitorTests.java [hunk 1]
```diff
@@ -164,10 +164,18 @@ private void doTestShape(Geometry geometry, GeometryDocValueReader reader, LatLo
     }
 
     private boolean isIdenticalPoint(Geometry geometry, LatLonGeometry latLonGeometry) {
-        if (geometry instanceof org.elasticsearch.geometry.Point point) {
-            if (latLonGeometry instanceof Point latLonPoint) {
+        if (latLonGeometry instanceof Point latLonPoint) {
+            if (geometry instanceof org.elasticsearch.geometry.Point point) {
                 return encodeLatitude(point.getLat()) == encodeLatitude(latLonPoint.getLat())
                     && encodeLongitude(point.getLon()) == encodeLongitude(latLonPoint.getLon());
+            } else if (geometry instanceof org.elasticsearch.geometry.Line line) {
```
[the remark this answers is not in the document]
Comment (iverase) [reply x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/LatLonGeometryRelationVisitorTests.java:171]: I think this is enough, I thought in the same fix.

**Review by iverase — APPROVED** [review]
lgtm

**Comment by iverase** [comment]
Can we backport this fix?

**Comment by craigtaverner** [comment]
Shall we back-port it to 8.7.0 and 8.6.3?

**Pull request merged**